thrift.go

```diff
@@ -0,0 +1,169 @@
+		return nil, err
+	}
+	return secretsStore, nil
+}
+
+func initTracing(cfg ServerConfig, logger log.Wrapper, metrics *metricsbp.Statsd) error {
+	if err := tracing.InitGlobalTracer(tracing.TracerConfig{
+		ServiceName:      cfg.Tracing.Namespace,
+		SampleRate:       cfg.Tracing.SampleRate,
+		Logger:           logger,
+		MaxRecordTimeout: cfg.Tracing.RecordTimeout,
+		QueueName:        "main",
+	}); err != nil {
+		return err
+	}
+
+	tracing.RegisterCreateServerSpanHooks(
+		metricsbp.CreateServerSpanHook{Metrics: metrics},
+		tracing.ErrorReporterCreateServerSpanHook{},
+	)
+	return nil
+}
+
+func initSentry(cfg ServerConfig) error {
+	if err := raven.SetDSN(cfg.Sentry.DSN); err != nil {
+		return err
+	}
+	if err := raven.SetSampleRate(float32(cfg.Sentry.SampleRate)); err != nil {
+		return err
+	}
+	raven.SetEnvironment(cfg.Sentry.Environment)
+	return nil
+}
+
+func cleanup(closers []io.Closer) {
+	for _, c := range closers {
+		if err := c.Close(); err != nil {
+			log.Errorw("msg", "err", err)
+		}
+	}
+}
+
+// NewThriftServer returns a server that initializes and includes the default
+// middleware and cross-cutting concerns needed in order to be a standard Baseplate service.
+//
+// At the moment, this includes secrets management, metrics, edge contexts
+// (edgecontext.InjectThriftEdgeContext), and spans/tracing (tracing.InjectThriftServerSpan).
+func NewThriftServer(ctx context.Context, cfg ServerConfig, processor thriftbp.BaseplateProcessor, additionalMiddlewares ...thriftbp.ProcessorMiddleware) (srv Server, err error) {
+	var closers []io.Closer
+	defer func() {
+		if err != nil {
+			cleanup(closers)
+		}
+	}()
+
+	initLogger(cfg)
+	logger := log.ZapWrapper(cfg.Log.Level)
+	bootstrapLogger := log.ZapWrapper(log.ErrorLevel)
+
+	metricsbp.M = metricsbp.NewStatsd(ctx, metricsbp.StatsdConfig{
+		CounterSampleRate:   &cfg.Metrics.CounterSampleRate,
+		HistogramSampleRate: &cfg.Metrics.HistogramSampleRate,
+		Prefix:              cfg.Metrics.Namespace,
+		Address:             cfg.Metrics.Endpoint,
+		LogLevel:            cfg.Log.Level,
+	})
+	closers = append(closers, metricsbp.M)
+
+	secretsStore, err := initSecrets(ctx, cfg, bootstrapLogger)
+	if err != nil {
+		return nil, err
+	}
+	closers = append(closers, secretsStore)
+
+	ecImpl := edgecontext.Init(edgecontext.Config{Store: secretsStore})
+	if err = initTracing(cfg, bootstrapLogger, metricsbp.M); err != nil {
+		return nil, err
+	}
+	closers = append(closers, opentracing.GlobalTracer().(*tracing.Tracer))
+	if err = initSentry(cfg); err != nil {
+		return nil, err
+	}
+	innerCfg := thriftbp.ServerConfig{
+		Addr:    cfg.Addr,
+		Timeout: cfg.Timeout,
+		Logger:  logger,
+	}
+
+	middlewares := []thriftbp.ProcessorMiddleware{
+		thriftbp.InjectServerSpan,
+		thriftbp.InjectEdgeContext(ecImpl),
+	}
+	middlewares = append(middlewares, additionalMiddlewares...)
+
+	ts, err := thriftbp.NewServer(innerCfg, processor, middlewares...)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This should already be handled by `thritbp.NewBaseplateServer` or whatever we called it?

Reply:
I think we only have [`thriftbp.NewServer`](https://pkg.go.dev/github.com/reddit/baseplate.go/thriftbp?tab=doc#NewServer), which does not add default middlewares. We can certainly add a `thriftbp.NewBaseplateServer` to pull in them, though.